app/src/main/java/com/keylesspalace/tusky/adapter/NotificationsAdapter.java

```diff
@@ -488,11 +499,58 @@ private void setupContentAndSpoiler(NotificationViewData.Concrete notificationVi
             Spanned content = statusViewData.getContent();
             List<Emoji> emojis = statusViewData.getStatusEmojis();
 
-            Spanned emojifiedText = CustomEmojiHelper.emojifyText(content, emojis, statusContent);
+            if(contentCollapseButton != null && statusViewData.isCollapsible() && (notificationViewData.isExpanded() || !hasSpoiler)) {
+                contentCollapseButton.setOnCheckedChangeListener((buttonView, isChecked) -> {
+                    int position = getAdapterPosition();
+                    if(position != RecyclerView.NO_POSITION && notificationActionListener != null) {
+                        notificationActionListener.onNotificationContentCollapsedChange(isChecked, position);
+                    }
+                });
+
+                contentCollapseButton.setVisibility(View.VISIBLE);
+                if(statusViewData.isCollapsed()) {
+                    contentCollapseButton.setChecked(true);
+                    statusContent.setFilters(new InputFilter[]{(source, start, end, dest, dstart, dend) -> {
```

Comment:
Please, extract this

Reply:
Would it be ok if I were to move it to SmartLengthFilter in the util package?
Also will I need to convert that to Kotlin or is Java ok? I don't have much experience with Kotlin.

Reply:
I implemented it this way, so once I'm done complaining about how bad it works on devices it shouldn't even run on (Moto G 2013), I (or someone else) will start working on the smart cutting features I discussed in [the comment below](https://github.com/tuskyapp/Tusky/pull/825#issuecomment-417800580).